code/mathematical_algorithms/src/tower_of_hanoi/tower_of_hanoi_iterative.c

```diff
@@ -0,0 +1,20 @@
+#include <stdio.h>
+#include <math.h>
+
+void toh (int n) {
+    int steps, start, end;
+    steps = pow (2, n);    // Minimum number of moves required
+    for (int i = 1; i < steps; i ++) {
```

Comment:
Declare looping variables above the loop for compatibility with older C compilers